app/components/candidate_interface/new_reference_history_component.rb

```diff
@@ -0,0 +1,29 @@
+module CandidateInterface
+  class NewReferenceHistoryComponent < ViewComponent::Base
+    attr_reader :reference
+
+    def initialize(reference)
+      @reference = reference
+    end
+
+    def history
+      ReferenceHistory.new(reference).all_events
+    end
+
+    def formatted_date(event)
+      event.time.to_fs(:govuk_date)
```

Comment:
I'd also just call this inline in the ERB template and do away with this method. 
Incidentally it looks like you're passing `event.time` from the templare but expecting an `event`.

Reply:
👍